src/parcsr_ls/par_interp_device.c

```diff
@@ -610,7 +610,7 @@ hypre_BoomerAMGBuildDirInterp_getcoef( HYPRE_Int   nr_of_rows,
    HYPRE_Real diagonal = 0.0, sum_N_pos = 0.0, sum_N_neg = 0.0, sum_P_pos = 0.0, sum_P_neg = 0.0;
 
    /* diag part */
-   HYPRE_Int p_diag_A, q_diag_A, p_diag_P, q_diag_P;
+   HYPRE_Int p_diag_A = 0, q_diag_A, p_diag_P = 0, q_diag_P;
```

Comment:
Just out of curiosity, is there a reason for initializing these variables? It seems they are set later and the initialized value is not used. Maybe I'm missing something...